flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/table/DataGeneratorConnectorITCase.java

```diff
@@ -69,6 +73,33 @@ public void testTypes() throws Exception {
             }
         }
 
-        Assert.assertEquals("Unexpected number of results", 10, results.size());
+        assertThat(results.size()).isEqualTo(10);
+    }
+
+    @Test
+    public void testLimit() throws Exception {
+        final TestingTableEnvironment env = TestingTableEnvironment
+                .create(
+                        EnvironmentSettings.newInstance().inStreamingMode().build(),
+                        null,
+                        new TableConfig());
+
+        env.executeSql("CREATE TABLE datagen_t (\n"
+                + "	f0 CHAR(1)\n"
+                + ") WITH ("
+                + "	'connector' = 'datagen'"
+                + ")");
+
+        List<Row> results = new ArrayList<>();
+
+        try (CloseableIterator<Row> iter = env
+                .executeSql("select * from datagen_t limit 5")
```

Comment:
This test actually still doesn't test the limit pushdown support of this source as the test cannot distinguish that from the limit applied generically over all sources.

Reply:
Maybe we can inffer that from it's logical plan.